### PR TITLE
Add Material Icons and Colors to completions (#454).

### PR DESCRIPTION
**Pull request by pq**
image

image

🎉 🎉 🎉 

Fixes: #454.

### CODE REVIEW

File: resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -280,4 +280,10 @@
 
     <library.type implementation="io.flutter.sdk.FlutterPluginLibraryType"/>
   </extensions>
+  
+  <!-- Dart Plugin extensions -->
```
Comment (devoncarew) [inline resources/META-INF/plugin.xml]: For this, you'll want to edit `plugin_template.xml` and regen using `bin/plugin generate`.

File: src/io/flutter/editor/FlutterColors.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  @Nullable
+  public static FlutterColor getColor(@NotNull String key) {
+    // Handle things like Colors.blue.shade200; convert the text to blue[200].
+    if (key.contains(".shade")) {
+      key = key.replace(".shade", "[") + "]";
+    }
+
+    if (colors.containsKey(key)) {
+      final Color color = getColorValue(key);
+      assert color != null;
```
Comment (devoncarew) [inline src/io/flutter/editor/FlutterColors.java]: we may not want to assert here, or below

File: src/io/flutter/editor/FlutterCompletionContributor.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import org.dartlang.analysis.server.protocol.CompletionSuggestion;
+import org.dartlang.analysis.server.protocol.Element;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.util.Objects;
+
+
+public class FlutterCompletionContributor extends com.jetbrains.lang.dart.ide.completion.DartCompletionExtension {
```
Comment (devoncarew) [inline src/io/flutter/editor/FlutterCompletionContributor.java]: add `com.jetbrains.lang.dart.ide.completion.DartCompletionExtension` to the imports?

File: resources/META-INF/plugin.xml (hunk 1 above)
Comment (pq) [reply resources/META-INF/plugin.xml]: 👍 Forgot about that.  Thanks!

File: src/io/flutter/editor/FlutterColors.java (hunk 2 above)
Comment (pq) [reply src/io/flutter/editor/FlutterColors.java]: As per our conversation, moved to if-checks.

File: src/io/flutter/editor/FlutterCompletionContributor.java (hunk 3 above)
Comment (pq) [reply src/io/flutter/editor/FlutterCompletionContributor.java]: Whoops.  Left over from some early fiddles.  Fixed.

**Comment by pq** [comment]
Build is green; ready for a proper review! 😄 

/cc @devoncarew

**Comment by devoncarew** [comment]
Looks great!

When we vette the next release, we need to test the versions well (AS 3.0, AS 3.1 + IntelliJ 2017.3, IntelliJ 2018.1) to make sure there aren't any issues with the jar versions and compilation.

**Comment by devoncarew** [comment]
Having the icons in there will be fantastic!

**Pull request merged**